UN-OCHA/unocha-site
# [UNO-631] Stories paragraph type

Refs: UNO-598, UNO-630, UNO-631

This PR does the following:

**News and Stories view**

1. Adds the drupal/verf module to be able to use a dropdown for the region and response filters
2. Updates the filters with a OR condition for the country, response and region to better it the view all links (see below, point 4 of stories paragraph type)

Notes:
- The "Stories" paragraph type replaces the blocks of the "News and Stories" view and should be removed when we clean up things.

**Stories paragraph type**

1. Adds a "Stories" paragraph type with a "title" (defaults to "Latest News and Stories") and a limit field (defaults to 3)
2. Adds the `unocha_paragraphs` module with notably a preprocess hook for the "stories" paragraph type with the logic to get the stories to display based on the parent of the paragraph (ex: response, etc.). Stories are ordered by "sticky" then "promote" then "created" date.
3. Adds a template override for the "stories" paragraph type to wrap the story articles and use the `uno-card-list` library.
4. Generates "view all" links to the filtered "News and Stories" page. @left23 Not sure if that's what we want but I though it's a better UX.

Notes:
- On a "response" page: show stories tagged with the response and complete with stories from the region if less than the limit
- On a "region" page: show stories tagged with the region
- On other node pages: use any "country", "response" or "region" field to filter the list or return the unfiltered list of stories (ex: homepage).
- The above point means, it's possible to add a "stories" paragraph type to a `story` page to show related stories if that's something we want to do at some point.
- **view modes**: currently the `card` view mode is hardcoded in the stories paragraph type preprocess. If we want to offer other view modes, we could add a mapping between the view mode of the stories paragraph type and the view mode of the story node.

### Deployment

After deployment, the existing content list paragraph types using the stories view should be replaced by a `stories` paragraph type.

### Tests

1. Checkout the branch and recreate the image or run `composer install` to install the `drupal/verf` module
2. Clear the cache and import the config
3. Add a "stories" paragraph  to a response, region and home page and check that the proper stories are displayed.
